### ISSUE TITLE
Bug in PCA 3D for tetrahedra

### ISSUE THREAD

**Issue by palliez** [CGAL/cgal#3684]
## Issue Details

The fitting plane is undefined for a union of tets that are well defined!!
when the tag dimension is 3.
when the dimension is 2 it works.. 

## Source Code
```c++
Point a(0.0, 0.0, 0.0);
Point b(0.0, 1.0, 0.0);
Point c(1.0, 1.0, 0.0);
Point d(1.0, 0.0, 0.0);
Point e(0.5, 0.5, 4.0);
Tetrahedron tet1(a, c, b, e);
Tetrahedron tet2(a, c, d, e);
std::list<Tetrahedron> tets;
tets.push_back(tet1);
tets.push_back(tet2);

Plane plane;
CGAL::linear_least_squares_fitting_3(tets.begin(), tets.end(), plane, CGAL::Dimension_tag<3>());
Vector n = plane.orthogonal_vector();
std::cout << "n: " << n << std::endl;
std::cout << "plane: " << plane << std::endl;
```
## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): windows
* Compiler: Visual C++ 2017
* Release or debug mode: release
* Specific flags used (if any):
* CGAL version: 4.13
* Boost version: 1,67
* Other libraries versions if used (Eigen, TBB, etc.): eigen 3

**Comment by sgiraudot** [comment]
Ok, this is worse than what I originally thought: I think I can safely say that PCA for Tetrahedra never worked in CGAL. 

I already found 2 major bugs due to the fact that `Tetrahedron_3::volume()` is considered as unsigned in the PCA package, whereas it really isn't (so there should be a `CGAL::abs()` around it):
* The error is done in `assemble_covariance_matrix_3()` for Tetrahedron
* It's also done in `CGAL::centroid()` for Tetrahedron (yes, even the centroid of a range of tetrahedra did not work and could produce `nan` coordinates…), which is used for assembling the covariance matrix too.

What worries me is: I fixed these errors, the `nan` are thus gone and the returned plane has valid attibrutes… but if we display the plane, it clearly _isn't_ the PCA of the tetrahedra (even when reducing to a single tetrahedron very elongated/flat, the plane does not seem related at all to it). So my guess is, there is something wrong in the way we compute the covariance matrix.
